fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -2330,7 +2330,19 @@ private String getAbsoluteUri(String baseUri, String relativeUri) {
             if (HTTPReturnPreference.REPRESENTATION.equals(FHIRRequestContext.get().getReturnPreference())) {
                 bundleEntryBuilder.resource(resource);
             } else if (HTTPReturnPreference.OPERATION_OUTCOME.equals(FHIRRequestContext.get().getReturnPreference())) {
-                bundleEntryBuilder.resource(operationOutcome);
+                OperationOutcome responseEntryOutcome = (OperationOutcome)responseEntry.getResource();
+                if (responseEntryOutcome != null && !responseEntryOutcome.equals(FHIRUtil.ALL_OK)) {
+                    if (operationOutcome.equals(FHIRUtil.ALL_OK)) {
+                        bundleEntryBuilder.resource(responseEntryOutcome);
```

Comment:
slight question here, should ALL_OK still be added along with the warning... Technically it's acceptable.

Reply:
the way it's currently working in doCreate and doPatchOrUpdate is that the OperationOutcome is generated by the FHIRUtil.buildOperationOutcome(issues) method, passing in any warnings found during the create or update. In that method, the ALL_OK is only added if there are no warnings. So that's the logic I used - will only return the ALL_OK if there are no warnings.

Reply:
Works for me,